File: search-server/src/main/java/com/example/search/repository/ProductSearchRepository.java
package com.example.search.repository;

import com.example.search.domain.Product;
import org.springframework.data.elasticsearch.repository.ElasticsearchRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface ProductSearchRepository extends ElasticsearchRepository<Product, String> {
    List<Product> findByNameContaining(String name);
    List<Product> findByCategory(String category);
}

File: search-server/src/main/kotlin/com/example/search/repository/ProductSearchRepository.kt
package com.example.search.repository

import com.example.search.domain.Product
import org.springframework.data.elasticsearch.repository.ElasticsearchRepository
import org.springframework.stereotype.Repository

@Repository
interface ProductSearchRepository : ElasticsearchRepository<Product, String> {
    fun findByNameContaining(name: String): List<Product>
    fun findByCategory(category: String): List<Product>
}
